schema.go

```diff
@@ -106,3 +113,34 @@ func findBuildByBuilder(db *mgo.Database, builder string, buildnum int) (*LogKee
 	}
 	return build, nil
 }
+
+func UpdateFailedTest(db *mgo.Database, id bson.ObjectId) error {
```

Comment:
Since we're considering the db to be global state, it's awkward to pass that around. I would drop the db parameter and look it up in each function that needs it